can/interfaces/ics_neovi/neovi_bus.py

```diff
@@ -0,0 +1,255 @@
+
+        for device in devices:
+            if serial is None:
+                dev = device
+                break
+            if str(device.SerialNumber) == serial:
+                dev = device
+                break
+        else:
+            msg = ['No device']
+
+            if type_filter is not None:
+                msg.append('with type {}'.format(type_filter))
+            if serial is not None:
+                msg.append('with serial {}'.format(serial))
+            msg.append('found.')
+            raise Exception(' '.join(msg))
+        ics.open_device(dev)
+        return dev
+
+    def _process_msg_queue(self, timeout=None):
+        try:
+            messages, errors = ics.get_messages(self.dev, False, timeout)
+        except ics.RuntimeError:
+            return
+        for ics_msg in messages:
+            if ics_msg.NetworkID != self.network:
+                continue
+            if ics_msg.ArbIDOrHeader == 0:
+                # Looks like ICS device sends frames with ArbIDOrHeader = 0
+                # Need to find out exactly what they are for
+                # Filtering them for now
+                continue
+            if not self._is_filter_match(ics_msg.ArbIDOrHeader):
+                continue
+            self.rx_buffer.put(ics_msg)
+        if errors:
+            logger.warning("%d errors found" % errors)
+
+            for msg in ics.get_error_messages(self.dev):
+                logger.warning(msg)
+
+    def _is_filter_match(self, arb_id):
+        """
+        If SW filtering is used, checks if the `arb_id` matches any of
+        the filters setup.
+
+        :param int arb_id:
+            CAN ID to check against.
+
+        :return:
+            True if `arb_id` matches any filters
+            (or if SW filtering is not used).
+        """
+        if not self.sw_filters:
+            # Filtering done on HW or driver level or no filtering
+            return True
+        for can_filter in self.sw_filters:
+            if not (arb_id ^ can_filter['can_id']) & can_filter['can_mask']:
+                return True
+        return False
+
+    def _get_timestamp_for_msg(self, ics_msg):
+        if self._use_system_timestamp:
+            # This is the system time stamp.
+            # TimeSystem is loaded with the value received from the timeGetTime
+            # call in the WIN32 multimedia API.
+            #
+            # The timeGetTime accuracy is up to 1 millisecond. See the WIN32
+            # API documentation for more information.
+            #
+            # This timestamp is useful for time comparing with other system
+            # events or data which is not synced with the neoVI timestamp.
+            #
+            # Currently, TimeSystem2 is not used.
+            return ics_msg.TimeSystem
+        else:
+            # This is the hardware time stamp.
+            # The TimeStamp is reset to zero every time the OpenPort method is
+            # called.
```

Comment:
Since the timestamp is zeroed on OpenPort I see a possibility to create an offset using `time.time()` and add that to the timestamp to have an absolute timestamp.

Reply:
Will add this

Reply:
For that the user should use http://python-ics.readthedocs.io/en/latest/#ics.set_rtc. The "OpenPort" is not necessarily called only called by us. The driver allow multiple clients.

Do you think we should set the RTC when we open the device? Should we provide a API in this class toto set the RTC?

Reply:
I'm not familiar with how the NEOVI interfaces work so I'll leave that up to you to decide.